packages/medusa/src/migrations/1612359285299-product_option_product_id.ts

```diff
@@ -0,0 +1,29 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class productOptionProductId1612359285299 implements MigrationInterface {
+  name = "productOptionProductId1612359285299"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "product_option" DROP CONSTRAINT "FK_e634fca34f6b594b87fdbee95f6"`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "product_option" ALTER COLUMN "product_id" SET NOT NULL`
+    )
+    await queryRunner.query(
+      `COMMENT ON COLUMN "product_option"."product_id" IS NULL`
+    )
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `COMMENT ON COLUMN "product_option"."product_id" IS NULL`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "product_option" ALTER COLUMN "product_id" DROP NOT NULL`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "product_option" ADD CONSTRAINT "FK_e634fca34f6b594b87fdbee95f6" FOREIGN KEY ("product_id") REFERENCES "product"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`
+    )
+  }
+}
```

Comment:
Looks like this removes the foreign key constraint i.e. referential integrity is broken. Not sure why it generates it like that, but I think the current models will work without this as the `product_option` table already has foreign key with the correct name: `product_id`

Reply:
Yeah to be honest.. This migration is looking really weird. We really only need `ALTER TABLE "product_option" ALTER COLUMN "product_id" SET NOT NULL` as far as I can see. Correct me if I am wrong? @srindom

Reply:
I believe TypeORM adds a `NOT NULL` constraint by default to the foreign keys created. So the migration is unnecessary altogether.

Reply:
But the column `product_id` is missing. Before this migration, we were not able to do `productOptionRepo.create({ ...someOption, product_id: product.id })`